### PR TITLE
Add ``DeprecationWarning`` for non-string ``current_name`` & ``file``

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

Turned out to be quite easy.

@Pierre-Sassoulas I wonder if this breaks anything. I don't think it shouldn't, but I'm hesitant about it...

### CODE REVIEW

File: tests/lint/unittest_lint.py [hunk 1]
```diff
@@ -251,7 +251,7 @@ def reporter():
 @pytest.fixture
 def init_linter(linter: PyLinter) -> PyLinter:
```
Comment (Pierre-Sassoulas) [inline tests/lint/unittest_lint.py:252]: I don't like this fixture name, should be a noun like `initialized_linter` ? But it's outside of the scope of this MR.

**Review by Pierre-Sassoulas — APPROVED** [review]
:ok_hand:

**Comment by DanielNoord** [comment]
What do you think about possible breakage of external tools? As shown if you `enable` or `disable` on the `linter` object directly this can break stuff... However, I'm not sure if that is supported in the first place.

File: tests/lint/unittest_lint.py (hunk 1 above)
Comment (DanielNoord) [reply tests/lint/unittest_lint.py:252]: I'll do a follow-up.

**Comment by Pierre-Sassoulas** [comment]
>  possible breakage of external tools

Yeah, this will potentially breaks plugins that did not give the file prevousely. We need a deprecation period first, and we can only merge in 3.0.

**Comment by DanielNoord** [comment]
Difficult to deprecate to.. Not sure how we could warn people.

**Comment by Pierre-Sassoulas** [comment]
> Not sure how we could warn people.

I'm thinking a condition in ``MessageLocation`` or ``PyLinter.add_message`` like:

```python
if not abspath: # if not self.current_file: ...
    warning.warn("'current_file' will need to be a valid path in 3.0...")
```

File: tests/test_check_parallel.py [hunk 2]
```diff
@@ -297,7 +297,6 @@ def test_sequential_checkers_work(self) -> None:
 
         # Add a sequential checker to ensure it records data against some streams
         linter.register_checker(SequentialTestChecker(linter))
-        linter.enable("R9999")
```
Comment (Pierre-Sassoulas) [inline tests/test_check_parallel.py:300]: 👍

**Review by Pierre-Sassoulas — COMMENTED** [review]
LGTM but maybe we need some tests for the deprecation warning ?

**Comment by Pierre-Sassoulas** [comment]
Also should it be in 2.13 as it's only deprecation warning and no actual breaking changes ?

**Comment by DanielNoord** [comment]
> LGTM but maybe we need some tests for the deprecation warning ?

We could. Imo, not really necessary. I don't think this is warning against any real code. There is so much depending on `current_name` being set, so this probably only raises against testing code.



> Also should it be in 2.13 as it's only deprecation warning and no actual breaking changes ?

Yeah, the milestone is from the previous version!

**Review by Pierre-Sassoulas — APPROVED** [review]

**Review by Pierre-Sassoulas — APPROVED** [review]
I let this go stale when it could be merged, my bad.

**Comment by DanielNoord** [comment]
No worries! I often leave stuff open after I or you have reviewed to give others a chance to review but forget to revisit at a later point.

**Comment by Pierre-Sassoulas** [comment]
>  I often leave stuff open after I or you have reviewed to give others a chance to review

Same here. As long as there's not too much conflicts there's no downside and we can merge a bunch of approved MR when we close up a release.

**Pull request merged**